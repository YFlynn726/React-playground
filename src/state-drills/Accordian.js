import React from 'react';

class Accordian extends React.Component {
    static defaultProps = {tabs: []};
    state = {
        currentSectionIndex: 0
      };
    
      
    handleButtonClick(index) {
        this.setState({ currentSectionIndex: index })
      }
    
      renderButtons() {
        return this.props.tabs.map((tab, index) => (
          <button key={index} onClick={() => this.handleButtonClick(index)}>
            {tab.title}
          </button>
        ))
      }
      renderContent() {
        const currentSection = this.props.tabs[this.state.currentSectionIndex]
        return (
          <p className='content'>
            {currentSection.content}
          </p>
        )
      }
    
    render() {
        return (
        <ul>
            <li>
            {this.renderButtons()}
          {this.props.tabs.length && this.renderContent()}
            </li>
          
        </ul>
        )
    }
}

export default Accordian