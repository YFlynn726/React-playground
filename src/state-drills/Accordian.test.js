import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Accordian from './Accordian';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`Accordian Component`, () => {
  // array of objects, each with a name and content
  const sectionsProp = [
    { name: 'First tab',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.' },
    { name: 'Second tab',
      content: 'Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit.' },
    { name: 'Third tab',
      content: 'Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur.' },
  ];
  
  it('renders the first tab by default', () => {
    const tree = renderer.create(<Accordian tabs={sectionsProp} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('closes the first tab and opens any clicked tab', () => {
    const wrapper = shallow(<Accordian tabs={sectionsProp} />)
    console.log('>>> WRAPPER <<<')
    console.log(wrapper.debug())
    console.log('>>> FIND(BUTTON) <<<')
    console.log(wrapper.find('button').at(1).debug())
    wrapper.find('button').at(1).simulate('click')
    console.log('>>> WRAPPER <<<')
    console.log(wrapper.debug())
    console.log('>>> FIND(BUTTON) <<<')
    console.log(wrapper.find('button').at(1).debug())
    
  })

it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordian />, div);
    ReactDOM.unmountComponentAtNode(div);
})

it('renders empty given no tabs', () => {
  const wrapper = shallow(<Accordian />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders the first tab by default', () => {
  const wrapper = shallow(<Accordian tabs={sectionsProp} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
})