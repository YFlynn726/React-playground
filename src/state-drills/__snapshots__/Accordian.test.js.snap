// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component renders empty given no tabs 1`] = `
<ul>
  <li>
    0
  </li>
</ul>
`;

exports[`Accordian Component renders the first tab by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    />
    <button
      onClick={[Function]}
    />
    <button
      onClick={[Function]}
    />
    <p
      className="content"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.
    </p>
  </li>
</ul>
`;

exports[`Accordian Component renders the first tab by default 2`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    />
    <button
      key="1"
      onClick={[Function]}
    />
    <button
      key="2"
      onClick={[Function]}
    />
    <p
      className="content"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque.
    </p>
  </li>
</ul>
`;
